src/store/actions/auth.js

```diff
@@ -130,13 +162,104 @@ export const authGo = (email, password, isRegister) => {
   }
 };
 
+export const authSetProfileGo = (token, displayName, photoUrl, isEdit) => {
+  return (dispatch) => {
+    const apiKey = process.env.REACT_APP_FIREBASE_API_KEY;
+
+    const authData = {
+      idToken: token,
+      displayName: displayName,
+      photoUrl: photoUrl
+    }
+
+    let url = `https://identitytoolkit.googleapis.com/v1/accounts:update?key=${apiKey}`;
+
+    axios.post(url, authData)
+      .then((response) => {
+        dispatch(authDoneLoading());
+        dispatch(closeModal());
+        setTimeout(() => {
+          dispatch(openAndSetActiveModalAndMessage('success', `Your profile has been ${isEdit ? 'updated' : 'set'}!`));
+        }, 250);
+        dispatch(authGetProfile(token));
+      }).catch((error) => {
+        let errorMessage = '';
+        if(error.response) {
+          errorMessage = normalizeErrorString(error.response.data.error.message);
+        } else {
+          errorMessage = error.message;
+        }
+
+        dispatch(authFail(errorMessage));
+        dispatch(closeModal());
+        setTimeout(() => {
+          dispatch(openAndSetActiveModalAndMessage('error', errorMessage));
+        }, 250);
+      });
+  }
+}
+
+export const authSetProfile = (token, displayName, photoUrl, isEdit) => {
+  return (dispatch) => {
+    dispatch(authStart());
+
+    const photoUpdated = typeof(photoUrl) === 'object';
+    if(photoUpdated) {
+      const ref = firebaseStorageRef.child(photoUrl.name);
+
+      ref.put(photoUrl).then((response) => {
```

Comment:
You should be able to chain these, as the `ref.someMethod` calls return promises, as far as I understand.

```suggestion
      ref.put(photoUrl)
	.then((putResponse) => {
		return ref.getDownloadURL();
	})
	.then((getUrlResponse) => {
		dispatch(authSetProfileGo(token, displayName, getUrlResponse, isEdit));
	});
```

Reply:
yep, it works. I didn't realize you could return the response like that!